ere-guests/stateless-validator/sp1/src/main.rs

```diff
@@ -46,3 +47,38 @@ fn init_tracing_just_like_println() {
         .with_max_level(tracing::Level::INFO) // capture info! and up
         .init(); // set as global default
 }
+
+fn from(fork_spec: ForkSpec) -> ChainSpec {
+    let spec_builder = ChainSpecBuilder::default()
+        .chain(Chain::mainnet())
+        .genesis(Genesis::default());
+
+    match fork_spec {
+        ForkSpec::Frontier => spec_builder.frontier_activated(),
+        ForkSpec::Homestead | ForkSpec::FrontierToHomesteadAt5 => {
+            spec_builder.homestead_activated()
+        }
+        ForkSpec::EIP150 | ForkSpec::HomesteadToDaoAt5 | ForkSpec::HomesteadToEIP150At5 => {
+            spec_builder.tangerine_whistle_activated()
+        }
+        ForkSpec::EIP158 => spec_builder.spurious_dragon_activated(),
+        ForkSpec::Byzantium
+        | ForkSpec::EIP158ToByzantiumAt5
+        | ForkSpec::ConstantinopleFix
+        | ForkSpec::ByzantiumToConstantinopleFixAt5 => spec_builder.byzantium_activated(),
+        ForkSpec::Istanbul => spec_builder.istanbul_activated(),
+        ForkSpec::Berlin => spec_builder.berlin_activated(),
+        ForkSpec::London | ForkSpec::BerlinToLondonAt5 => spec_builder.london_activated(),
+        ForkSpec::Merge
+        | ForkSpec::MergeEOF
+        | ForkSpec::MergeMeterInitCode
+        | ForkSpec::MergePush0 => spec_builder.paris_activated(),
+        ForkSpec::Shanghai => spec_builder.shanghai_activated(),
+        ForkSpec::Cancun => spec_builder.cancun_activated(),
+        ForkSpec::ByzantiumToConstantinopleAt5 | ForkSpec::Constantinople => {
+            panic!("Overridden with PETERSBURG")
+        }
+        ForkSpec::Prague => spec_builder.prague_activated(),
+    }
+    .build()
```

Comment:
I found something interesting which IMHO feels a bit dangerous in how Reth works:
- The `ChainSpec` builder always assume [`None` for the deposit contract address](https://github.com/paradigmxyz/reth/blob/812dd04b807d689f4661651d32d021dfb50dc38f/crates/chainspec/src/spec.rs#L978).
- I was wondering how this could even work for EEST tests, and found that in their EIP-6110 they [have a default rule](https://github.com/alloy-rs/evm/blob/main/crates/evm/src/eth/eip6110.rs#L107).

I double checked with Mario and all EEST tests always assume mainnet deposit contract address, which probably explains why they did this -- but still I would make the `build()` panic/error if an explicit address isn't provided. In any case, this comment is unrelated to this PR.